velox/connectors/hive/HiveConnector.cpp

```diff
@@ -60,7 +60,7 @@ HiveConnector::HiveConnector(
                     SimpleLRUCache<std::string, std::shared_ptr<FileHandle>>>(
                     hiveConfig_->numCacheFileHandles())
               : nullptr,
-          std::make_unique<FileHandleGenerator>(nullptr)),
+          std::make_unique<FileHandleGenerator>(config)),
```

Comment:
Would it be possible to add a test case?

Should we change FileHandleGenerator to not allow not-specifying config or specifying null config?

CC: @xiaoxmeng

Reply:
I add a not null check in FileSystem ctor, I think it can protect derived class access config safely.

Reply:
FileHandleGenerator will need a null config for local filesystem and a not-null config for other filesystems.  That is why it has both constructors.
```
 FileHandleGenerator() {}
 FileHandleGenerator(std::shared_ptr<const Config> properties)
```

Reply:
ignore above wrong msg.

Reply:
ignore above wrong msg.

Reply:
> FileHandleGenerator() {}

In the `FileHandleGenerator() {}` can we make a empty config and set it to `properties_` instead? In this way, we can get ride of nullptr

Reply:
Fix current issue is easy, just pass a valid config is ok, but how do we avoid similar issue in the future? such as `std::make_unique<FileHandleGenerator>()),`.

Regardless of whether the filesystem needs a config, only allow passing a config to FileHandleGenerator is safe.